### PR TITLE
Show descriptive error message when submitting too long text submissions

### PR DESCRIPTION
**Pull request by joschahenningsen**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all newly inserted strings into English and German.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).

### Motivation and Context

When submitting text exercises longer than 30,000 characters the error message didn't hint on what the issue was.

### Description
I changed the error to be a BadRequest error because it makes more sense than a 500. Also I made it throw an exception that shows it's message to the client.

### Steps for Testing

1. Submit a long text submission to a text exercise (> 30,000 characters). (I copied this https://de.wikipedia.org/wiki/Metallocene)
2. observe that a message appears that sufficiently indicates what the issue is.

### Screenshots

image

### CODE REVIEW

**Review by ndricimrr — DISMISSED** [review]
Not a PR yet, but Tested on TS2. Worked well 👍

File: src/main/webapp/app/exercises/text/participate/text-editor.component.ts [hunk 1]
```diff
@@ -239,8 +239,8 @@ export class TextEditorComponent implements OnInit, OnDestroy, ComponentCanDeact
                     this.jhiAlertService.warning('entity.action.submitDeadlineMissedAlert');
                 }
             },
-            () => {
-                this.jhiAlertService.error('artemisApp.modelingEditor.error');
+            (err) => {
+                this.jhiAlertService.error(err.error.message);
```
Comment (FrankeLukas) [inline src/main/webapp/app/exercises/text/participate/text-editor.component.ts:243]: Can we make sure the error message is actually set (to what we expect)? How does this look/work when an error different to the one specific BadRequest happens?

Comment (joschahenningsen) [reply src/main/webapp/app/exercises/text/participate/text-editor.component.ts:243]: All errors we throw in this resource seem to have an associated message(with translations). Here's an example of something generic going wrong: 
image
If you want I can add a check nonetheless.

Comment (FrankeLukas) [reply src/main/webapp/app/exercises/text/participate/text-editor.component.ts:243]: If all other errors have messages with translations this is fine. Thanks for clarifying! 👍

File: src/main/webapp/app/exercises/text/participate/text-editor.component.ts [hunk 2]
```diff
@@ -239,8 +239,8 @@ export class TextEditorComponent implements OnInit, OnDestroy, ComponentCanDeact
                     this.jhiAlertService.warning('entity.action.submitDeadlineMissedAlert');
                 }
             },
-            () => {
-                this.jhiAlertService.error('artemisApp.modelingEditor.error');
+            (err) => {
```
Comment (FrankeLukas) [inline src/main/webapp/app/exercises/text/participate/text-editor.component.ts:242]: Bonus points if you add a type here.

**Review by FrankeLukas — APPROVED** [review]
Code changes lgtm.

File: src/main/webapp/app/exercises/text/participate/text-editor.component.ts (hunk 2 above)
Comment (joschahenningsen) [reply src/main/webapp/app/exercises/text/participate/text-editor.component.ts]: done, thanks!

**Review by FrankeLukas — APPROVED** [review]
Re-approve.

File: src/main/webapp/i18n/de/error.json [hunk 3]
```diff
@@ -34,6 +34,7 @@
         "feedbackCreditsNull": "Fehler! Wenn Feedback vorhanden ist, muss das Feedback eine Punktzahl haben.",
         "invalidGradeStepFormat": "Eine oder mehrere Notenstufen folgen nicht dem richtigen Format. Klicke auf das \"?\" für mehr Informationen.",
         "invalidGradeStepAdjacency": "Die Notenstufen bilden keinen gültigen Notenschlüssel ab. Klicke auf das \"?\" für mehr Informationen.",
-        "emptyGradeSteps": "Die Notenstufen dürfen nicht leer sein. Klicke auf das \"?\" für mehr Informationen."
+        "emptyGradeSteps": "Die Notenstufen dürfen nicht leer sein. Klicke auf das \"?\" für mehr Informationen.",
+        "textSubmissionTooLong": "Text Abgaben können höchstens 30000 Zeichen lang sein."
```
Comment (krusche) [inline src/main/webapp/i18n/de/error.json:38]:
```suggestion
        "textSubmissionTooLong": "Text Abgaben können höchstens 30.000 Zeichen lang sein."
```

**Review by krusche — APPROVED** [review]

**Pull request merged**